src/ipa_pc/mod.rs

```diff
@@ -672,21 +725,45 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
             let hiding_bound = labeled_polynomial.hiding_bound();
             let commitment = labeled_commitment.commitment();
 
-            combined_polynomial += (cur_challenge, polynomial);
-            combined_commitment_proj += &commitment.comm.mul(cur_challenge);
+            let p_len = polynomial.coeffs.len();
+
+            if p_len > key_len {
+
+                let mut polynomial_lc = Polynomial::zero();
+                for i in 0..p_len / key_len + if p_len % key_len != 0 { 1 } else { 0 } {
+                    let is = i * key_len;
+                    let poly_single = Polynomial::from_coefficients_slice(&polynomial.coeffs[i * key_len..core::cmp::min((i + 1) * key_len, p_len)]);
+                    polynomial_lc += (point.pow(&[is as u64]), &poly_single);
+                }
+                combined_polynomial += (cur_challenge, &polynomial_lc);
+
+                let mut comm_lc = <G::Projective as ProjectiveCurve>::zero();
+                for (i, comm_single) in commitment.comm.iter().enumerate() {
+                    let is = i * key_len;
+                    comm_lc += &comm_single.mul(point.pow(&[is as u64]));
+                }
+                combined_commitment_proj += &comm_lc.mul(&cur_challenge);
+
+            } else {
```

Comment:
As above: I expect the code in the if clause to work also for the case p_len <= key_len.

Reply:
As above: you do additional instructions (for no reason)

Reply:
I'd say that style is a reason? In terms of performance not much is changed.

Reply:
Already redone